### PR TITLE
refactor(chunkserver): Give own files to ChunkserverEntry

### PR DESCRIPTION
**Pull request by lgsilva3087**
[Must be merged after PR #210]

ChunkserverEntry now has its own .h and .cc files, which cleans up the
NetworkWorkerThread related files.

Misc changes in network_worker_thread.h and .cc:
- Remove unneeded includes.
- Use constexpr instead of macros.
- Wrap long lines.

### CODE REVIEW

File: src/chunkserver/chunkserver_entry.cc [hunk 1]
```diff
@@ -0,0 +1,1578 @@
...
+			}
+			return;
+		}
+		stats_bytesin += bytesRead;
+		fwdInputPacket.startPtr += bytesRead;
+		fwdInputPacket.bytesLeft -= bytesRead;
+		if (fwdInputPacket.bytesLeft > 0) {
+			return;
+		}
+		ptr = fwdHeaderBuffer + 4;
```
Comment (rolysr) [inline src/chunkserver/chunkserver_entry.cc:1236]: Please consider specifying a variable naming the 4 value for making clear why is this value used

File: src/chunkserver/chunkserver_entry.cc [hunk 2]
```diff
@@ -0,0 +1,1578 @@
...
+
+	outPacket->outputBuffer = getReadOutputBufferPool().get(sizeOfWholePacket);
+
+	if (outPacket->outputBuffer->copyIntoBuffer(packetPrefix) !=
+	    static_cast<ssize_t>(packetPrefix.size())) {
+		if (outPacket->outputBuffer) {
+			getReadOutputBufferPool().put(std::move(outPacket->outputBuffer));
+		}
+
+		return nullptr;
```
Comment (rolysr) [inline src/chunkserver/chunkserver_entry.cc:158]: Consider setting a name for better readability on this nullptr return.

File: src/chunkserver/chunkserver_entry.cc [hunk 3]
```diff
@@ -0,0 +1,1578 @@
...
+// initialize connection to another CS
+int ChunkserverEntry::initConnection() {
+	TRACETHIS();
+	int status;
+	// TODO(msulikowski) If we want to use a ConnectionPool, this is the right
+	// place to get a connection from it
+	fwdSocket = tcpsocket();
+	if (fwdSocket < 0) {
+		safs_pretty_errlog(LOG_WARNING, "create socket, error");
+		return -1;
```
Comment (rolysr) [inline src/chunkserver/chunkserver_entry.cc:234]: Consider setting also an specific name for each one of these -1 returns according to the error.

File: src/chunkserver/chunkserver_entry.cc [hunk 4]
```diff
@@ -0,0 +1,1578 @@
...
+		connectStartTimeUSec = eventloop_utime();
+	}
+
+	return 0;
+}
+
+void ChunkserverEntry::retryConnect() {
+	TRACETHIS();
+	tcpclose(fwdSocket);
+	fwdSocket = -1;
```
Comment (rolysr) [inline src/chunkserver/chunkserver_entry.cc:266]: Here could also be considered to set `fwdSocket` to some `invalidFwdSocket`.

File: src/chunkserver/chunkserver_entry.cc [hunk 5]
```diff
@@ -0,0 +1,1578 @@
...
+		}
+
+		todoReadCounter++;
+		offset += thisPartSize;
+		size -= thisPartSize;
+	}
+}
+
+void ChunkserverEntry::ping(const uint8_t *data, PacketHeader::Length length) {
+	if (length != 4) {
```
Comment (rolysr) [inline src/chunkserver/chunkserver_entry.cc:406]: Same here on considering naming the value 4 here for better readability after refactoring.

File: src/chunkserver/chunkserver_entry.cc [hunk 6]
```diff
@@ -0,0 +1,1578 @@
...
+}
+
+void ChunkserverEntry::generateChartPNGorCSV(const uint8_t *data,
+                                             uint32_t length) {
+	TRACETHIS();
+	uint32_t chartid;
+	uint8_t *ptr;
+	uint32_t len;
+
+	if (length != 4) {
```
Comment (rolysr) [inline src/chunkserver/chunkserver_entry.cc:947]: Same here, consider naming the value in this case

File: src/chunkserver/chunkserver_entry.cc [hunk 7]
```diff
@@ -0,0 +1,1578 @@
...
+	}
+}
+
+void ChunkserverEntry::generateChartData(const uint8_t *data, uint32_t length) {
+	TRACETHIS();
+	uint32_t chartid;
+	uint8_t *ptr;
+	uint32_t len;
+
+	if (length != 4) {
```
Comment (rolysr) [inline src/chunkserver/chunkserver_entry.cc:975]: This is another place to consider naming the value 4.

**Review by rolysr — COMMENTED** [review]
LGTM. Consider some cosmetic changes as this change is mostly for refactoring for better code maintanibility and readability

File: src/chunkserver/chunkserver_entry.cc [hunk 8]
```diff
@@ -0,0 +1,1578 @@
...
+
+/* IDLE operations */
+
+void ChunkserverEntry::hddListV2(const uint8_t *data, uint32_t length) {
+	TRACETHIS();
+	uint32_t opSize;
+	uint8_t *ptr;
+
+	(void)data;
+	if (length != 0) {
```
Comment (rolysr) [inline src/chunkserver/chunkserver_entry.cc:913]: Why different from 0 here?

File: src/chunkserver/network_worker_thread.cc [hunk 9]
```diff
@@ -18,1523 +18,51 @@
...
-NetworkWorkerThread::NetworkWorkerThread(uint32_t nrOfBgjobsWorkers, uint32_t bgjobsCount)
-		: doTerminate(false) {
+NetworkWorkerThread::NetworkWorkerThread(uint32_t nrOfBgjobsWorkers,
+                                         uint32_t bgjobsCount)
+    : doTerminate(false) {
 	TRACETHIS();
 	eassert(pipe(notify_pipe) != -1);
 #ifdef F_SETPIPE_SZ
-	eassert(fcntl(notify_pipe[1], F_SETPIPE_SZ, 4096*32));
+	eassert(fcntl(notify_pipe[1], F_SETPIPE_SZ, 4096 * 32));
```
Comment (antuan96314) [inline src/chunkserver/network_worker_thread.cc:62]: Consider adding a descriptive naming for magic constant `4096 * 32`

File: src/chunkserver/chunkserver_entry.h [hunk 10]
```diff
@@ -0,0 +1,367 @@
...
+	/// Callback for chunk block retrieval completion.
+	static void sauGetChunkBlocksFinishedCallback(uint8_t status, void *entry);
+	/// Callback for chunk block retrieval completion.
+	static void getChunkBlocksFinishedCallback(uint8_t status, void *entry);
+
+	/// Retrieves chunk blocks from the given information.
+	void getChunkBlocks(const uint8_t *data, uint32_t length);
+
+	/// Retrieves chunk blocks from the given information using the new way.
+	void sauGetChunkBlocks(const uint8_t *data, uint32_t length);
```
Comment (antuan96314) [inline src/chunkserver/chunkserver_entry.h:310]:
```suggestion
	void getChunkBlocks(const uint8_t *data, uint32_t length);
```
Consider removing the `sau` prefix to match other method naming convention inside the same object

File: src/chunkserver/chunkserver_entry.h [hunk 11]
```diff
@@ -0,0 +1,367 @@
...
+	                                                    void *entry);
+	/// Callback for chunk block retrieval completion.
+	static void sauGetChunkBlocksFinishedCallback(uint8_t status, void *entry);
+	/// Callback for chunk block retrieval completion.
+	static void getChunkBlocksFinishedCallback(uint8_t status, void *entry);
+
+	/// Retrieves chunk blocks from the given information.
+	void getChunkBlocks(const uint8_t *data, uint32_t length);
+
+	/// Retrieves chunk blocks from the given information using the new way.
```
Comment (antuan96314) [inline src/chunkserver/chunkserver_entry.h:309]: Consider adding some details to explain what is `the new way`

File: src/chunkserver/chunkserver_entry.h [hunk 12]
```diff
@@ -0,0 +1,367 @@
...
+	static void readFinishedCallback(uint8_t status, void *entry);
+	/// Callback after delayed close operations.
+	static void delayedCloseCallback(uint8_t status, void *entry);
+	/// Callback for when a write operation finishes.
+	static void writeFinishedCallback(uint8_t status, void *entry);
+	/// Callback for legacy chunk block retrieval completion.
+	static void sauGetChunkBlocksFinishedLegacyCallback(uint8_t status,
+	                                                    void *entry);
+	/// Callback for chunk block retrieval completion.
+	static void sauGetChunkBlocksFinishedCallback(uint8_t status, void *entry);
```
Comment (antuan96314) [inline src/chunkserver/chunkserver_entry.h:302]:
```suggestion
	static void getChunkBlocksFinishedCallback(uint8_t status, void *entry);
```
Consider removing the `sau` prefix to match other method naming convention inside the same object

File: src/chunkserver/chunkserver_entry.cc [hunk 13]
```diff
@@ -0,0 +1,1578 @@
...
+#include "common/massert.h"
+#include "protocol/SFSCommunication.h"
+#include "common/legacy_vector.h"
+#include "protocol/packet.h"
+#include "slogger/slogger.h"
+#include "common/sockets.h"
+#include "devtools/request_log.h"
+#include "devtools/TracePrinter.h"
+
+constexpr uint32_t kMaxPacketSize = 100000 + SFSBLOCKSIZE;
```
Comment (antuan96314) [inline src/chunkserver/chunkserver_entry.cc:60]: Consider maybe explaining why 100 000 plus SFSBLOCKSIZE is the maximum allowed Packet size

**Review by antuan96314 — APPROVED** [review]
I added some minor suggestions that are not stopper to merge. LGTM

**Review by rolysr — APPROVED** [review]

File: src/chunkserver/network_worker_thread.cc (hunk 9 above)
Comment (lgsilva3087) [reply src/chunkserver/network_worker_thread.cc:62]: Not in the scope of this change.
will be handled as part of the refactoring, but this change is only moving to new files to facilitate the review.

File: src/chunkserver/chunkserver_entry.h (hunk 12 above)
Comment (lgsilva3087) [reply src/chunkserver/chunkserver_entry.h:302]: Not in the scope of this change.
will be handled as part of the refactoring, but this change is only moving to new files to facilitate the review.

File: src/chunkserver/chunkserver_entry.h [hunk 14]
```diff
@@ -0,0 +1,367 @@
...
+	uint32_t offset = 0; ///< R: Offset within the chunk for the operation.
+	uint32_t size = 0; ///< R: Size of the current operation.
+
+	/// Pointer to the concrete serializer singleton.
+	/// Serializers coud be of type:
+	/// - LegacyMessageSerializer: for legacy messages
+	/// - SaunaFsMessageSerializer: for new messages
+	MessageSerializer* messageSerializer = nullptr; // R+W
+
+	LOG_AVG_TYPE readOperationTimer;
```
[the remark this answers is not in the document]
Comment (lgsilva3087) [reply src/chunkserver/chunkserver_entry.h:153]: Not in the scope of this change.
will be handled as part of the refactoring, but this change is only moving to new files to facilitate the review.

File: src/chunkserver/chunkserver_entry.cc (hunk 13 above)
Comment (lgsilva3087) [reply src/chunkserver/chunkserver_entry.cc:60]: Not in the scope of this change.
will be handled as part of the refactoring, but this change is only moving to new files to facilitate the review.

File: src/chunkserver/chunkserver_entry.h (hunk 11 above)
Comment (lgsilva3087) [reply src/chunkserver/chunkserver_entry.h:309]: It was answered here: https://github.com/leil-io/saunafs/pull/210#discussion_r1799270845
I will do it but in later commits, this one is moving the files without to many other changes to facilitate the review.
Please note that if I move the content to new files, the reference for the review is lost (side to side), making very difficult to review if there are also additional changes.

File: src/chunkserver/chunkserver_entry.h (hunk 10 above)
Comment (lgsilva3087) [reply src/chunkserver/chunkserver_entry.h:310]: It was answered here: https://github.com/leil-io/saunafs/pull/210#discussion_r1799270845

**Pull request merged**